### PR TITLE
issue 3: cut over to opensearch

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Moved (cut over) to opensearch instead of elasticsearch. Some refactoring took place to support the move.

## ⚙️ Test Data and/or Report
Integration tests pass:

## ♻️ Related Issues
#3

### CODE REVIEW

File: service/src/main/java/gov/nasa/pds/api/registry/search/HitIterator.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/search/HitIterator.java:13]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/registry/business/LidVidUtils.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/LidVidUtils.java:60]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java:63]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java:125]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/search/RegistrySearchRequestBuilder.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/search/RegistrySearchRequestBuilder.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java:122]: ignore

**Comment by al-niessner** [comment]
@jordanpadams @tloubrieu-jpl 

It now uses opensearch instead of elasticsearch. According to find, all elastic have been excised from the code.

Can we do a high priority fix to match the documentation (swagger.yml) with code? I would like to keep repo frozen after this branch merges in for the high priority fix. I should be done with it by Thursday but will be sending lots of emails for details to put in docs when I find discrepancies.

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: fixed

**Comment by al-niessner** [comment]
@jordanpadams @tloubrieu-jpl 

Can I get this reviewed so I can merge it please?

**Comment by tloubrieu-jpl** [comment]
@al-niessner sorry about the delay here, I am busy on a different project today and yesterday I was blocked by something not working right with the docker-compose deployment (which I thought could help me).

To unblock you you can create a branch off this branch.

**Comment by al-niessner** [comment]
Seems I can wait. I am blocked by gateway errors at swaggerhub more than this and there is nothing any of us can do about swaggerhub but wait.

**Review by tloubrieu-jpl — APPROVED** [review]
Thanks @al-niessner , that works

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

This PR is closed but not merged. Is there a reason for that?

**Comment by tloubrieu-jpl** [comment]
@al-niessner my bad, it was merged on my local repository (do you care ?) and not on github. And the pull request had a tag "merged" already, I don't understand why, maybe it wads only partially merged...

**Comment by al-niessner** [comment]
Yes, I am trying to get my next round of changes back to main and do not want a giant conflict so I would like this PR merged. Since it is in your repo then all you need do it push main and we should be good.

**Comment by jordanpadams** [comment]
@al-niessner your issue_3.1 branch was merged with main per above. can you check that you have the latest changes on main and make your updates on a new branch from there?

i can confirm your commits were merged to main per e824e39813ac785d1609872b632702491143effe , so unless someone removed them in a later commit, that should be done.

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Yes it is there now. Thanks.

**Pull request merged**
